### PR TITLE
Added managing network policy topic

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Based on:
- https://help.stackrox.com/docs/manage-network-policies/
- https://help.stackrox.com/docs/manage-network-policies/view-network-policies/
- https://help.stackrox.com/docs/manage-network-policies/simulate-network-policies/
- https://help.stackrox.com/docs/manage-network-policies/simulate-network-policies/
- https://help.stackrox.com/docs/manage-network-policies/use-network-baselining/

Preview: https://openshift-docs-git-manage-network-policies-gnelson.vercel.app/openshift-acs/master/operating/manage-network-policies.html

### CODE REVIEW

File: modules/apply-generated-policies.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
```
Comment (bergerhoffer) [inline modules/apply-generated-policies.adoc:1]: This file isn't included anywhere

File: modules/apply-generated-policies.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+[role="_abstract"]
+
+You can apply the generated network policies from the RHACS portal.
+
+.Procedure
+* To directly apply the generated policies in the cluster from within {product-title}, select *Apply Network Policies*.
+
+[WARNING]
+====
+Directly applying network policies may cause problems for running applications.
```
Comment (bergerhoffer) [inline modules/apply-generated-policies.adoc:17]: may -> might

File: modules/apply-generated-policies.adoc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Always download and test the network policies in development environment or testing clusters, before you apply them to production workloads.
```
Comment (bergerhoffer) [inline modules/apply-generated-policies.adoc:18]: * in development environment -> in a development environment ?
* testing -> test ?
* And comma could be removed

File: modules/configure-cidr-blocks.adoc [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// * operating/manage-network-policies.adoc
+:_module-type: PROCEDURE
+[id="configure-cidr-blocks_{context}"]
+= Configuring CIDR blocks
+
+[role="_abstract"]
+You can specify custom CIDR blocks or configure displaying auto-discovered CIDR block in the *Network Graph* view.
+
```
Comment (bergerhoffer) [inline modules/configure-cidr-blocks.adoc:10]: Needs the `.Procedure` heading

File: modules/delete-all-autogenerated-policies.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+You can delete all generated policies from your cluster that you have created by using {product-title}.
+
+.Procedure
+
+* Run the following command:
+** On {ocp}:
++
+[source,terminal]
+----
+$ oc get ns -o jsonpath='{.items[*].metadata.name}' | xargs -n 1 oc delete networkpolicies -l 'network-policy-generator.stackrox.io/generated=true' -n
```
Comment (bergerhoffer) [inline modules/delete-all-autogenerated-policies.adoc:18]: Just checking that the `-n` at the end of these commands isn't expecting something to be passed in (like the namespace).

File: modules/delete-generated-policies.adoc [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * operating/manage-network-policies.adoc
+:_module-type: PROCEDURE
+[id="delete-generated-policies_{context}"]
+= Deleting generated policies
+
+[role="_abstract"]
+If you have applied generated policies directly and want to remove them, select the *Revert most recently applied YAML* icon on the Network Policy Simulator panel.
```
Comment (bergerhoffer) [inline modules/delete-generated-policies.adoc:9]: "Network Policy Simulator" in bold?

File: modules/generate-network-policies.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+{product-title} allows you to autogenerate these network policies based on the actual observed network communication flows in your environment.
+
+You can generate network policies from the network graph view.
+
+The generated policies apply to the deployments shown in the network graph and they allow all network traffic observed during the selected time.
+
+.Procedure
+. In the RHACS portal, select *Network Graph* from the left-hand navigation menu.
+. Select a cluster name from the menu on the top bar, if the right one is not already selected.
+. If you want to generate policies for only some deployments, use the filter box to filter the deployments you are after.
```
Comment (bergerhoffer) [inline modules/generate-network-policies.adoc:23]: you are after -> you want

File: modules/network-graph-view.adoc [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+In the *Network Graph* view, you can configure the type of connections you want to see.
+In the *Flows* section (upper left), select:
+
+* *Active* to view only active connections.
+* *Allowed* to view only allowed network connections.
+* *All* to view both active and allowed network connections.
+
+In the network graph, each circle represents a deployment, each surrounding box represents a Kubernetes namespace, and each thick line represents connection between namespaces.
+You can move your mouse over these items to view more details.
```
Comment (bergerhoffer) [inline modules/network-graph-view.adoc:22]: "move your mouse over" -> "hover over"

File: modules/network-graph-view.adoc [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+To understand the meaning of other symbols in the network graph, move you mouse over other symbols in the legend (lower left) to view the tooltip indicating the meaning of those symbols.
+
+When you move your mouse over:
```
Comment (bergerhoffer) [inline modules/network-graph-view.adoc:26]: "move your mouse over" -> "hover over"

File: modules/view-network-policies.adoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+[role="_abstract"]
+Network policies specify how groups of pods are allowed to communicate with each other and with other network endpoints.
+Kubernetes `NetworkPolicy` resources use labels to select pods and define rules that specify what traffic is allowed to or from the selected pods.
+{product-title} discovers and displays network policy information for all your Kubernetes clusters, namespaces, deployments, and pods, in the *Network Graph* view.
+
+To view network policies and other related details for deployments in a namespace, you can select the namespace in *Network Graph* view.
+
+The namespace details panel lists all deployments in the selected namespace.
+You can then move your mouse over a deployment in the details panel and select the *Navigate to deployment* (arrow) icon that appears on the right to view deployment details.
```
Comment (bergerhoffer) [inline modules/view-network-policies.adoc:16]: "move your mouse over" -> "hover over"

File: modules/network-graph-view.adoc [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+The *Network Graph* view shows information about network connections between managed clusters and external sources.
+{product-title} also automatically discovers and highlights public CIDR (Classless Inter-Domain	Routing) addresses blocks, such as Google Cloud, AWS, Azure, Oracle Cloud, and Cloudflare.
+Using this information, you can identify deployments with active external connections and if they are making or receiving unauthorized connections from outside of your network.
+
+[NOTE]
+====
+You need {product-title} version 3.0.52 or newer to see information about active external connections.
+====
+
+By default, the external connections point to a common *External Entities* box, and different CIDR (Classless Inter-Domain Routing) addresses blocks in the *Network Graph* view.
```
Comment (bergerhoffer) [inline modules/network-graph-view.adoc:79]: You've already spelled out CIDR above, so I don't think you need to again here

File: modules/policy-generation-strategy.adoc [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+= Policy generation strategy
+
+When you autogenerate network policies:
+
+* {product-title} generates a single network policy for each deployment in the namespace.
+The pod selector for the policy is the pod selector of the deployment.
+** If a deployment already has a network policy, {product-title} does not generate new policies or delete existing policies.
+* Generated policies only restrict traffic to existing deployments.
+** Deployments you create later will not have any restrictions unless you create or generate new network policies for them.
+** If a new deployment needs to contact a deployment with a network policy, you may need to edit the network policy to allow access.
```
Comment (bergerhoffer) [inline modules/policy-generation-strategy.adoc:15]: may -> might

File: modules/save-generated-policies.adoc [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+[role="_abstract"]
+You can download and save the generated network policies from {product-title}.
+Use this option to commit the policies into a version control system like Git.
+
+.Procedure
+* Select the *Download YAML* icon on the *Network Policy Simulator* panel.
++
+[NOTE]
+====
+. To create policies using the saved YAML file, use the following command:
```
Comment (bergerhoffer) [inline modules/save-generated-policies.adoc:17]: I know I haven't been pointing out notes/admonitions with bullets and procedures, but this one is kind of unnecessarily long for the information it's actually conveying. Suggest condensing it.

File: modules/simulate-network-policies.adoc [hunk 14]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * operating/manage-network-policies.adoc
+:_module-type: PROCEDURE
+[id="simulate-network-policies_{context}"]
+= Simulating network policies
+
+[role="_abstract"]
+Your current network policies may allow unneeded network communications.
```
Comment (bergerhoffer) [inline modules/simulate-network-policies.adoc:9]: may -> might

File: modules/simulate-network-policies.adoc [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+. On the *Network Graph* view header, select *Network Policy Simulator*.
+. Select *Upload and simulate network policy YAML* and upload your proposed YAML file.
+The network graph view displays what your proposed network policies would achieve.
+. To share your proposed policies with your team, select *Share YAML*.
+To apply your policy directly, select *Apply Network Policies*.
+
+[WARNING]
+====
+Applying network policies directly can cause problems for running applications.
+Be sure to test policies in development or testing clusters before applying them to production workloads.
```
Comment (bergerhoffer) [inline modules/simulate-network-policies.adoc:23]: testing clusters -> test clusters ?

File: modules/view-network-policies.adoc [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+
+Alternatively, you can directly select a deployment in the *Network Graph* view to view its details.
+The deployment details panel includes the *Network Flows*, *Details*, and *Network Policies* tabs.
+
+You can select each tab to view related information.
```
Comment (bergerhoffer) [inline modules/view-network-policies.adoc:21]: Need a blank line after this for the below to render properly as bullets.

File: modules/delete-all-autogenerated-policies.adoc (hunk 5 above)
Comment (gaurav-nelson) [reply modules/delete-all-autogenerated-policies.adoc:18]: Thanks @bergerhoffer Yes, I confirmed, the command is correct as it is. The namespace is passed by `xargs`.

File: modules/view-network-policies.adoc [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+To view network policies and other related details for deployments in a namespace, you can select the namespace in *Network Graph* view.
+
+The namespace details panel lists all deployments in the selected namespace.
+You can then hover over a deployment in the details panel and select the *Navigate to deployment* (arrow) icon that appears on the right to view deployment details.
+
+Alternatively, you can directly select a deployment in the *Network Graph* view to view its details.
+The deployment details panel includes the *Network Flows*, *Details*, and *Network Policies* tabs.
+
+You can select each tab to view related information.
+* The *Network Flows* tab shows information about ingress and egress connections, protocols, and port numbers in use for that deployment.
```
Comment (bergerhoffer) [inline modules/view-network-policies.adoc:22]: This still needs a blank line before these bullets in order for them to render properly

File: modules/test-generated-policies.adoc [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+.Procedure
+. To create policies using the saved YAML file, use the following command:
++
+[source,terminal]
+----
+$ oc create -f "<generated_file>.yaml" <1>
+----
+<1> Use `kubectl` instead of `oc` if you are using Kubernetes.
+. If the generated policies cause problems, you can remove them by running the following command:
+[source,terminal]
```
Comment (bergerhoffer) [inline modules/test-generated-policies.adoc:19]: Need a + above the code block in order for it to be indented properly.

**Review by bergerhoffer — COMMENTED** [review]
Two last formatting issues, then a conflict to resolve, then this can be merged!

File: modules/test-generated-policies.adoc [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+= Testing generated policies
+
+After you download the network policies that {product-title} generates, you can test them by applying them to your cluster.
+
+.Procedure
+. To create policies using the saved YAML file, use the following command:
++
+[source,terminal]
+----
+$ oc create -f "<generated_file>.yaml" <1>
```
Comment (bergerhoffer) [inline modules/test-generated-policies.adoc:15]: Won't hold up the PR over it, but the file name here shouldn't be in quotes. If you can consider updating in a future PR

**Comment by bergerhoffer** [comment]
LGTM

**Pull request merged**